### ISSUE TITLE
Please add more instruction

### ISSUE THREAD

**Issue by thomaspavelka** [codearchitects/GraphQLTools#3]
Hello!

Thank you for providing this tool. Unfortunately, from the provided image alone it is not clear to me how to achieve syntax highlighting. Could you add more instruction? Thank you very much!

Best, Thomas

**Comment by thomaspavelka** [comment]
Ok, I figured it out after all. It could have been faster though... You can use it via the constructor of GraphQLRequest as follows:

![grafik](https://github.com/user-attachments/assets/bd9eced4-27b9-4b4e-a7c2-3f627976be7e)

        public static GraphQLRequest? GetLatestIssueIDQuery()
        {
            return new GraphQLRequest(
                @"
                query($includeTitle: Boolean!) {
                    repository(owner: ""tpavelka"", name: ""Test"") {
                        issues(first: 1, orderBy: {field: CREATED_AT, direction: DESC}) {
                            edges {
                                node {
                                    id
                                    title @include(if: $includeTitle)
                                }
                            }
                        }
                    }
                }",
                variables: new { includeTitle = false }
            );
        }

**Comment by ABIOLI-COM** [comment]
It's not working anyway, for me, Visual Studio 2022 Community 17.12.4

**Comment by hasse112** [comment]
Yea it's not working for me neither.
There is no mention of where to specify which schema to use - this extension is a waste of time.

I find it amusing/tragic how the developer had time to write a couple of sentences about StringSyntax being case insensitive but not about how to actually use the extension...

**Comment by DeFlight** [comment]
Yep,. doesn't work for me either.  Looks like the dev doesn't look at these messages?

How he hell do you get this to work?

**Comment by thomaspavelka** [comment]
Did you guys try it the way I posted it above? It's working well for me the way I posted it

**Comment by DeFlight** [comment]
`var request = new GraphRequest()
{
    Query = """
        query getOrders($email: String!) {                               
            orders(first: 50, query: $email) {
                edges {
                    node {
                        createdAt
                        customer{
                            email
                        }

                    }
                }
            }
        }
        """,
    Variables = new Dictionary<string, object> { { "email", $"email:{email}" } }
};`

This does not do anything for me.

Am I ,missing something?

<img width="422" alt="Image" src="https://github.com/user-attachments/assets/79390c84-623e-499c-b682-631e136ea781" />

**Comment by DeFlight** [comment]
<img width="532" alt="Image" src="https://github.com/user-attachments/assets/03d6e330-80c9-4025-8406-94c138faf098" />

<img width="641" alt="Image" src="https://github.com/user-attachments/assets/b9f6a457-bdf2-485d-91df-9753f22eff98" />

Both of these do not work for me.  Am I missing something?

**Comment by DeFlight** [comment]
<img width="325" alt="Image" src="https://github.com/user-attachments/assets/a80d89bb-1717-43ef-966c-d9a9c5267f20" />

**Comment by thomaspavelka** [comment]
Please follow the pattern which I send above, then it should work. Let me know if it worked for you.

**Comment by DeFlight** [comment]
<img width="680" alt="Image" src="https://github.com/user-attachments/assets/7123d400-4da6-4543-9ebf-74c0a2cda9d4" />

Just put your exact code into my string input, can issue.

**Comment by thomaspavelka** [comment]
You don't seem to follow my example exactly

**Comment by DeFlight** [comment]
I am using a library that uses QL requests.  So maybe that is the whol eissue.

**Comment by thomaspavelka** [comment]
Just write a new function that returns the Request and follow the pattern I illustrated above. It should work

**Comment by DeFlight** [comment]
What library are you using?  As you can see, does not work.  Again, if you see something I am missing let me know.  As others said I can not get this to work.

<img width="651" alt="Image" src="https://github.com/user-attachments/assets/fa8eb39e-c5d8-45be-896a-0a8d36d2ce0d" />

**Comment by thomaspavelka** [comment]
I'll have a look tomorrow

**Comment by thomaspavelka** [comment]
I checked it again on a different setup. I used VS 17.10.5, downloaded and installed the [extension](https://marketplace.visualstudio.com/items?itemName=codearchitects-research.GraphQLTools), dowloaded and installed the [GraphQL.Client](https://github.com/graphql-dotnet/graphql-client) Nuget package in my solution, and inserted the code I pasted above, and it works flawlessly:

![Image](https://github.com/user-attachments/assets/8956552b-b0c8-4a1b-b664-a676209f0ea5)

Let me know, if this helps.

**Comment by DeFlight** [comment]
I will take a look.  I am using this as the library so I am wondering if that is the main issue on my end.

https://github.com/nozzlegear/ShopifySharp

**Comment by gianvitodifilippo** [comment]
Hello everyone, I'm a former Code Architects employee, and the main contributor to this repository.
I'm sorry your messages were unanswered but, as you can imagine, I'm no longer using the company GitHub account, so there was no chance for me of reading your messages.
In fact, I'm surprised (in a positive way) to see that somebody is actually using this extension!

That said, from what I remember, it's supposed to integrate with the `[StringSyntax]` attribute, so it should work on any field or parameter decorated with `[StringSyntax("graphql")]` without any further configuration. Apologies if the instructions provided in the readme are unclear.

By the way, I turned on notifications for this repository just in case. I remember that there were performance issues that needed to be fixed and more features to implement. If somebody is actually using it, I can reach out to my former company to see if we can keep the development going.

Cheers!

**Comment by bertgt** [comment]
> I will take a look. I am using this as the library so I am wondering if that is the main issue on my end.
> 
> https://github.com/nozzlegear/ShopifySharp

I'm (trying) to use this with ShopifySharp and can't get it to work.  So I am assuming ShopifySharp library has not implemented the required [StringSyntax("graphql")] attributes.

**Comment by gianvitodifilippo** [comment]
@bertgt can you provide a minimal code sample?

**Comment by bertgt** [comment]
Hi @gianvitodifilippo 

I have GraphQL tools 0.3 extension installed, and I've got graphql.config.yml and graphql.schema.json from the SopifySharp project included in my project's home folder.

FWIW VS 2022 infrequently complains the GraphQL Tools extension takes a while to load and offers to disable it. It's definitely loaded and enabled ;)

Here's a Graph QL query that I can't get intellisense for.

`var graphRequest = new GraphRequest
{
    Query =
    @"
    query listProducts($limit: Int!, $query: String!) 
    {
        products(first: $limit, query: $query) 
        {
            pageInfo 
            {
                startCursor
                endCursor
                hasNextPage
                hasPreviousPage
            }
            nodes 
            {
                createdAt
                id
                legacyResourceId
                description
                handle
                title
                hasOnlyDefaultVariant
                updatedAt
                variantsCount
                {
                    count
                }
                variants(first:250)
                  {
                    nodes
                    {
                      createdAt
                      displayName
                      id
                      legacyResourceId
                      title
                      updatedAt                                  
                    }
                  }
            }
        }
    }",
    Variables = new Dictionary<string, object>
    {
        {"limit", 250},
        {"query", "status:" + ProductStatus.ACTIVE}
    }
};`
